compiler/rustc_metadata/src/rmeta/decoder/cstore_impl.rs

```diff
@@ -248,6 +248,26 @@ provide! { tcx, def_id, other, cdata,
             .process_decoded(tcx, || panic!("{def_id:?} does not have trait_impl_trait_tys")))
      }
 
+    assoc_items_for_rpitits => {
+        cdata
+            .root
+            .tables
+            .assoc_items_for_rpitits
+            .get(cdata, def_id.index)
+            .map(|lazy| lazy.decode((cdata, tcx)))
+            .process_decoded(tcx, || panic!("{:?} does not have associated items for rpits", def_id))
+    }
+
+    impl_assoc_items_for_rpitits => {
```

Comment:
Additionally, does this query need an external provider, or should we just leave it unimplemented for extern crates? It should only be needed for the `associated_items` query, right (which already is cached in crate metadata), if I understand correctly.

Reply:
Did you meant to refer to `rpitit_associated_item` here?, because we are already encoding `assoc_items_for_rpitits` and `impl_assoc_items_for_rpitits`.

Reply:
No, I mean why are we encoding the impl RPITIT items? Where is that query going to be used for foreign impls?

wouldn't those items be accessed via the regular associated_items query, which is encoded separately?

Reply:
To explain my thinking more -- we *do* need to encode the trait RPITIT items, so we know which ones to synthesize when we see a local impl for a foreign trait. But for foreign impls, we don't need to do anything with those RPITIT items?